diktat-rules/src/test/resources/test/paragraph4/null_checks/IfConditionNullCheckExpected.kt

```diff
@@ -47,3 +47,26 @@ value
 }
 }
 
+fun foo() {
+    var result: Int? = 10
+    while (result != 0 ) {
+        result?.let {
+goo()
+}
+?: run {
+for(i in 1..10)
+break
+}
+    }
+    while (result != 0) {
+        result = goo()
+        result?.let {
+goo()
+}
+?: run {
+println(123)
+return
```

Comment:
This will retunr from `foo`, but the original intent was to break from `while`

Reply:
Fixed